src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1069,3 +1074,38 @@ export const disableSmsVerificationsForUser = (
       return transformMongoError(error)
     },
   ).map(() => true)
+
+/**
+ * Checks if the given form field should be updated.
+ * This currently checks if the admin has exceeded their free sms limit.
+ * @param form The form which the specified field belongs to
+ * @param formField The field which we should perform the update for
+ * @returns ok(form) If the field can be updated
+ * @return err(PossibleDatabaseError) if an error occurred while performing the required checks
+ * @return err(SmsLimitExceededError) if the form admin went over the free sm limit
+ * while attempting to toggle verification on for a mobile form field
+ */
+export const shouldUpdateFormField = (
```

Comment:
now that I'm reading this function signature again, should we make it clearer so ppl reading only the controller can understand what this function does?

Can we further split this function up so it can be extended if needed?

```ts
const shouldUpdateFormField = (...) => {
  switch (formField.fieldType) {
    case BasicField.Mobile:
      return assessMobileFieldUpdateAllowed(...)
    default:
      return okAsync(form)
}
```
`assessMobileFieldUpdateAllowed` can still be a function in this admin-form service since it's still acting on form fields.

If not, we can also move this field type check into the controller and call `assessMobileFieldUpdateAllowed`. This makes the pipelines clearer

Reply:
i was thinking about doing the split-up but was wary about premature optimization actually HAHA now that you've raised it, i'll split it up!

Reply:
done [here](https://github.com/opengovsg/FormSG/pull/2300/commits/8b5d20c5df364df3daed5e9e41fba411be333248)